core/src/main/kotlin/org/evomaster/core/search/gene/NumberGene.kt

```diff
@@ -25,4 +33,36 @@ abstract class NumberGene<T : Number>(name: String, var value: T) : Gene(name, m
             value.toFloat()
 
 
+    fun modifyValue(randomness: Randomness, value: Double, delta: Double, maxRange: Long, specifiedJumpDelta: Int, precisionChangeable: Boolean): Double{
+        val strategies = ModifyStrategy.values().filter{
+            precisionChangeable || it != ModifyStrategy.REDUCE_PRECISION
+        }
+        return when(randomness.choose(strategies)){
+            ModifyStrategy.SMALL_CHANGE-> value + min(1, maxRange) * delta
+            ModifyStrategy.LARGE_JUMP -> value + specifiedJumpDelta * delta
+            ModifyStrategy.REDUCE_PRECISION -> BigDecimal(value).setScale(randomness.nextInt(15), RoundingMode.HALF_EVEN).toDouble()
+        }
+
+    }
+
+    /**
+     * @return formatted value
```

Comment:
this comment just states the name of the method... :)
you can clarify what a "formatted value" is